psi4/src/psi4/fisapt/fisapt.cc

```diff
@@ -308,44 +331,53 @@ void FISAPT::partition() {
         ZCp[indC[ind]] = mol->Z(indC[ind]);
     }
 
+    vectors_["ZA_orig"]->copy(*vectors_["ZA"]);
+    vectors_["ZB_orig"]->copy(*vectors_["ZB"]);
+    vectors_["ZC_orig"]->copy(*vectors_["ZC"]);
+
     // => Local Orbital Targets <= //
 
     std::vector<int> orbsA;
     std::vector<int> orbsB;
     std::vector<int> orbsC;
+    std::vector<int> orbsL;
 
     // => Assign Links <= //
 
     std::string link_assignment = options_.get_str("FISAPT_LINK_ASSIGNMENT");
-    if (!(link_assignment == "AB" || link_assignment == "C")) {
+    if (!(link_assignment == "AB" || link_assignment == "C" || link_assignment == "SAO0" || link_assignment == "SAO1" || link_assignment == "SAO2" || link_assignment == "SIAO0" || link_assignment == "SIAO1" || link_assignment == "SIAO2" )) {
```

Comment:
As a possible thought for cleanup, the entire set of `link_assignment == SAOx/SIAOx` comparisons could probably be collapsed cleanly using regex. Something like:

```
bool is_sao = std::regex_match(link_assignment, std::regex("S[I]?AO[0-2]"));

if (!(link_assignment == "AB" || link_assignment == "C" || is_sao))
 ```
 
Might need an `#include <regex>` in the header list if it is not already included via another header file.

Not necessary by any means, but maybe something worth considering.

Reply:
I know I'm old :) so the current code looks more readable to me, but if the youngsters want this change, I'll be happy to make it...